hayeon/programmers-2/삼각-달팽이.js

```diff
@@ -0,0 +1,42 @@
+function solution(n) {
+  const arr = Array.from({ length: n }, (_, index) => Array(index + 1).fill(0));
+  let answer = [];
+  let count = 0;
+  let currentX = -1;
+  let currentY = 0;
+
+  while (n > 0) {
+    for (let i = 0; i < n; i++) {
+      currentX++;
+      count++;
+      arr[currentX][currentY] = count;
+    }
+    for (let i = 0; i < n - 1; i++) {
+      currentY++;
+      count++;
+      arr[currentX][currentY] = count;
+    }
+    for (let i = 0; i < n - 2; i++) {
+      currentX--;
+      currentY--;
+      count++;
+      arr[currentX][currentY] = count;
+    }
+    n -= 3;
+  }
```

Comment:
```suggestion
  for (let i = n; i > 0; i -= 3) {
    for (let j = 0; j < i; j++) {
      currentX++;
      count++;
      arr[currentX][currentY] = count;
    }
    for (let j = 0; j < i - 1; j++) {
      currentY++;
      count++;
      arr[currentX][currentY] = count;
    }
    for (let j = 0; j < i - 2; j++) {
      currentX--;
      currentY--;
      count++;
      arr[currentX][currentY] = count;
    }
  }
```
- 해당 코드는 for문으로 작성할 수 있습니다